.github/workflows/test.yml

```diff
@@ -31,4 +31,4 @@ jobs:
           hatch run dev:cov
       - name: Upload Coverage to Codecov
         if: matrix.os == 'ubuntu-latest' && matrix.python-version == 3.9
-        uses: codecov/codecov-action@v2
+        uses: codecov/codecov-action@v5
```

Comment:
FYI codecov v5 requires setting `CODECOV_TOKEN` as an environment variable and the upload job will fail silently if the token is not available.